mokapot/brew.py

```diff
@@ -1,21 +1,45 @@
 """
 Defines a function to run the Percolator algorithm.
 """
+
 import logging
 import copy
+from operator import itemgetter
 
-import pandas as pd
 import numpy as np
 from joblib import Parallel, delayed
+from typeguard import typechecked
 
 from .model import PercolatorModel
+from . import utils
+from .dataset import (
+    LinearPsmDataset,
+    calibrate_scores,
+    update_labels,
+)
+from .parsers.pin import parse_in_chunks
+from .constants import (
+    CHUNK_SIZE_ROWS_PREDICTION,
+    CHUNK_SIZE_READ_ALL_DATA,
+)
 
 LOGGER = logging.getLogger(__name__)
 
 
 # Functions -------------------------------------------------------------------
-def brew(psms, model=None, test_fdr=0.01, folds=3, max_workers=1, rng=None):
-    """Re-score one or more collection of PSMs.
+@typechecked
```

Comment:
Although, not all parameters of the function have type hints yet, the decorator will raise a TypeError if `brew` is used with type-hinted parameters that don't match the type hints. It's not necessary but it might be helpful to users.

Reply:
Humm I am not all that familiar with using runtime typechecked ... would it be worth it to have a test that actually checks that it fails when a wrong type is passed?

Reply:
Hmm not sure about that. Doing so would test functionality of @typchecked rather than Mokapot, so I would argue such a test is hopefully already living in the typechecked repo. In another discussion (and other location it was brought up, that we actually have tests that check the dtypes (although they are checked with the @typechecked decorator). I would rather remove those tests, so that we stay consistent what to test.

I hope this is ok for you.